### DISCUSSION TITLE
ds.rio.write_crs('epsg:4326', inplace=True) gives an error

### DISCUSSION THREAD

**Discussion by shuai-zhou** [corteva/rioxarray#447] [Q&A]
I did some tests on my Windows machine, everything works fine. But when I was working on a Linux-based HPC, the following line:

```python
ds.rio.write_crs('epsg:4326', inplace=True)
```

gives an error:

`ERROR 1: PROJ: proj_create_from_database: conda_env_name/share/proj/proj.db lacks DATABASE.LAYOUT.VERSION.MAJOR / DATABASE.LAYOUT.VERSION.MINOR metadata. It comes from another PROJ installation.`

This error did not stop the script, but when clipping a dataset using `clipped = ds.rio.clip(geodf.geometry.apply(mapping), geodf.crs)`, another error occurs:

`CRS not found. Please set the CRS with 'rio.write_crs()'. Data variable: precipitation`

Probably because the first error (`Error 1: PROJ: proj_create_from_database...`) did not write a CRS? How can I fix the first error? Thanks.

**Comment by snowman2** [comment]
For the first error, it seems GDAL is having issues finding the PROJ database that is expected. I would check the PROJ_LIB environment variable.

**Reply by shuai-zhou** [reply]
I set up the environment in the beginning of the script,

```python
import os
os.environ['PROJ_LIB'] = 'conda_env_name/share/proj'
os.environ['GDAL_DATA'] = 'conda_env_name/share/gdal'
```

There is still an error:

`ERROR 1: PROJ: proj_create_from_database: Cannot find proj.db`

**Reply by shuai-zhou** [reply]
maybe I specify the wrong path?

**Reply by snowman2** [reply]
I would recommend specifying the full path. You are currently specifying a relative path.

**Reply by snowman2** [reply]
Instead of the relative path: `conda_env_name/share/proj`, I recommend using the full path, for example: `/home/username/conda_directory/conda_env_name/share/proj`.

**Reply by shuai-zhou** [reply]
I changed the 'PROJ_LIB' to `'~/storage/home/my_account/work/envs/spatial/share/proj/proj.db'`, is this what you mean about full path? the error `ERROR 1: PROJ: proj_create_from_database: Cannot find proj.db` persists

**Reply by snowman2** [reply]
Can you change it to: `/storage/home/my_account/work/envs/spatial/share/proj` ?

**Reply by shuai-zhou** [reply]
I remembered you mentioned somewhere that do not try `pip install` in a conda environment. So I checked my `pip list`, and there was another `rioxarray`, I uninstalled it using `pip uninstall rioxarray`.

Then without specifying the two environment variables, `ds.rio.write_crs('epsg:4326', inplace=True)` still gave the same error, but I was able to run `clipped = ds.rio.clip(geodf.geometry.apply(mapping), geodf.crs)` successfully. Maybe `ERROR 1: PROJ: proj_create_from_database: conda_env_name/share/proj/proj.db lacks DATABASE.LAYOUT.VERSION.MAJOR / DATABASE.LAYOUT.VERSION.MINOR metadata. It comes from another PROJ installation.` is not a fatal error in this situation?

**Reply by snowman2** [reply]
Sounds like the database worked, even if it wasn't the version PROJ expected. It could be a fatal error depending on how PROJ is used and if there are breaking changes between database. But, it sounds like you got lucky in this scenario.

**Reply by shuai-zhou** [reply]
Yeah, thanks for your time and patience. I also asked for help from the HPC support team, if there are some other system-specific settings that could solve this issue, I will post their solution here.
